bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -16,26 +16,66 @@
 
 package org.pitest.pitclipse.ui.view;
 
+import java.io.File;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.IToolBarManager;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.SWTError;
 import org.eclipse.swt.browser.Browser;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.ui.IActionBars;
 import org.eclipse.ui.part.ViewPart;
 
-import java.io.File;
-
 /**
  * A view displaying the HTML report generated by PIT.
  */
 public class PitView extends ViewPart implements SummaryView {
+    public static final String VIEW_ID = "org.pitest.pitclipse.ui.view.PitView";
+    public static final String BLANK_PAGE = "about:blank";
+    public static final String BACK_BUTTON_TEXT = "<";
+    public static final String HOME_BUTTON_TEXT = "Home";
+    public static final String FORWARD_BUTTON_TEXT = ">";
     private Browser browser = null;
+    private String homeUrlString = null;
 
     @Override
     public synchronized void createPartControl(Composite parent) {
         try {
             browser = new Browser(parent, SWT.NONE);
+            resetBrowser();
             browser.addProgressListener(new PitUiUpdatePublisher(browser));
+            // create back button
+            final Action backButtonAction = new Action(BACK_BUTTON_TEXT) {
+                @Override
+                public void run() {
+                    browser.back();
+                }
+            };
+            // create home button for navigation
+            final Action homeButtonAction = new Action(HOME_BUTTON_TEXT) {
+                @Override
+                public void run() {
+                    if (homeUrlString != null) {
+                        // only set url, if an home is set otherwise NPE
+                        browser.setUrl(homeUrlString);
+                    }
+                }
+            };
+            // create forward button
+            final Action forwardButtonAction = new Action(FORWARD_BUTTON_TEXT) {
+                @Override
+                public void run() {
+                    browser.forward();
+                }
+            };
+            IActionBars actionBars = getViewSite().getActionBars();
+            IToolBarManager toolBar = actionBars.getToolBarManager();
+            toolBar.add(backButtonAction);
+            toolBar.add(homeButtonAction);
+            toolBar.add(forwardButtonAction);
+            actionBars.updateActionBars();
         } catch (SWTError e) {
             MessageBox messageBox = new MessageBox(parent.getShell(), SWT.ICON_ERROR | SWT.OK);
```

Comment:
I know this is not your code but since we're there: `SWTError` is an error and usually you're not supposed to catch errors, which are catastrophic anyway... I'd simply remove the try catch, @echebbi ? This will also avoid uncovered code.

Reply:
Not sure about that. I agree that errors shouldn't usually being caught but in this case it may make sense. `Browser`'s constructor is roughtly implemented as follows:

```java
this.webBrowser = BrowserFactory.createWebBrowser();
if (this.webBrowser == null) {
    // factory was unable to load WebKit
    throw new SWTError();
}
```

I guess the constructor throws an error to report that the issue comes from the execution environment (a library is missing) but I believe that it makes sense to catch the error so that we can report it nicely to the user. I just tried to broadcast the error and it's basically swallowed: some log is added to the _Error Log_ view and that's it. Not really helpful for a lambda user.

Actually, I believe that we should do better reporting:

- the MessageBox should say that it cannot open a web browser to display Pitest's HTML report (otherwise user may wonder why the view even tries to open a browser),
- the MessageBox should report the error's message to help troubleshooting,
- the PitView could display an error message instead of just remaining blank.

Does that sound reasonable? If you agree with that I've written then we should address this in another PR.